compiler/lexical.js

```diff
@@ -19,136 +16,140 @@
  * @param {string} text The input regular expression
  * @return {string|object} Returns a string that is an error message if failed to parse the expression,
  *                         otherwise returns an object which is the syntax tree.
+ *
+ * Edited from https://github.com/CyberZHG/toolbox/blob/gh-pages/js/lexical.js
  */
 function parseRegex(text) {
-    'use strict';
-    function parseSub(text, begin, end, first) {
-        var i,
-            sub,
-            last = 0,
-            node = {'begin': begin, 'end': end},
-            virNode,
-            tempNode,
-            stack = 0,
-            parts = [];
-        if (text.length === 0) {
-            return 'Error: empty input at ' + begin + '.';
+  "use strict";
+  function parseSub(text, begin, end, first) {
+    var i,
+      sub,
```

Comment:
We shouldn't have linting changes in a feature pr. Can you make a seperate pr with linting and linting on push hooks? We have a template for that in the main circom repo I think.